pageserver/src/walreceiver.rs

```diff
@@ -264,7 +264,9 @@ fn walreceiver_main(
                     )?;
 
                     if newest_segno - oldest_segno >= 10 {
-                        timeline.checkpoint()?;
+                        // FIXME: The layered repository performs checkpointing in a separate thread, so this
+                        // isn't needed anymore. Remove 'checkpoint' from the Timeline trait altogether? OR
+                        //timeline.checkpoint()?;
```

Comment:
Since we decided to keep rocksdb for a while, let's keep this with repository format check.
Something like `if repository_format == RepositoryFormat::RocksDb { timeline.checkpoint()?; }`

Reply:
done